Improve import batching options (#1058)
tenzir/vast: doc/cli/vast-import.md

for later export) all Suricata events from the Eve JSON file passed via standard
input.

> The import command batches parsed events in table slices. To control the
> batching, the following options are available:

Reviewer: ```suggestion
The import command parses events into table slices or *batches*.
The following options control the batching:
```

Reviewer: Looks like you went with the previous wording in your new commit?

Author: You are right, I overlooked this one apparently.

> - `import.batch-encoding`: Controls the encoding of table slices. Available

Reviewer: ```suggestion
- `import.batch-encoding`: Selects the encoding of table slices. Available
```

  options are `msgpack` (row-based) and `arrow` (column-based).

> - `import.batch-size`: Sets an upper bound for the size of every table slice.

Reviewer: ```suggestion
- `import.batch-size`: Sets an upper bound for the number of events per table slice.
```

>   A table slice is the unit that all components work on, causing this to be a
>   high impact tuning parameter. Decreasing the table slice size causes reduced
>   latency up to the point where messaging overhead becomes important, and
>   increasing it may improve overall performance. Note that this setting does not
>   mean that events are forwarded to the index and archive immediately upon
>   exceeding the given number of events, as the table slices themselves are
>   buffered again. Setting this option to 0 causes the table slice size to be
>   unbounded, leaving control of the batch size to other parameters.

Reviewer: ```suggestion
  Most components in VAST operate on table slices, which makes the table slice size a fundamental tuning knob on the spectrum of throughput and latency. Small table slices allow for shorter processing times, resulting in more scheduler context switches and a more balanced workload. However, the increased pressure on the scheduler comes at the cost of throughput. A large table slice size allows actors to spend more time processing a block of memory, but makes them yield less frequently to the scheduler. As a result, other actors scheduled on the same thread may have to wait a little longer. 
  
  The `batch-size` merely controls number of events per table slice, but not necessarily the number of events until a component forwards a batch to the next stage in a stream. The [CAF streaming framework](https://actor-framework.readthedocs.io/en/latest/Streaming.html) uses a credit-based flow-control mechanism to determine buffering of tables slices. Setting `batch-size` to 0 causes the table slice size to be "unbounded" and leaves it to other parameters to determine the actual table slice size.
```

> - `import.batch-timeout`: Sets a timeout for forwarding buffered table slices to
>   the index and archive, and can cause the `import.batch-size` parameter to be
>   underrun.

Reviewer: ```suggestion
- `import.batch-timeout`: Sets a timeout for forwarding buffered table slices to
  the importer. If the timeout fires before a table slice reaches `import.batch-size`, then the table slice will contain fewer events but ship immediately.
```

🗣️💣

> An optional filter expression allows for importing the relevant subset of

Reviewer: This paragraph now needs a bit gentler introduction.
